src/main/java/be/seeseemelk/mockbukkit/block/state/StructureMock.java

```diff
@@ -0,0 +1,257 @@
+	}
+
+	@Override
+	public @NotNull BlockVector getStructureSize()
+	{
+		return this.structureSize;
+	}
+
+	@Override
+	public void setStructureSize(@NotNull BlockVector vector)
+	{
+		Preconditions.checkNotNull(vector, "Vector cannot be null");
+		Preconditions.checkArgument(StructureMock.isBetween(vector.getBlockX(), 0, StructureMock.MAX_SIZE), "Structure Size (X) must be between 0 and " + StructureMock.MAX_SIZE);
+		Preconditions.checkArgument(StructureMock.isBetween(vector.getBlockY(), 0, StructureMock.MAX_SIZE), "Structure Size (Y) must be between 0 and " + StructureMock.MAX_SIZE);
+		Preconditions.checkArgument(StructureMock.isBetween(vector.getBlockZ(), 0, StructureMock.MAX_SIZE), "Structure Size (Z) must be between 0 and " + StructureMock.MAX_SIZE);
+		this.structureSize = new BlockVector(vector.getBlockX(), vector.getBlockY(), vector.getBlockZ());
+	}
+
+	@Override
+	public void setMirror(@NotNull Mirror mirror)
+	{
+		Preconditions.checkNotNull(mirror, "Mirror cannot be null");
+		this.mirror = mirror;
+	}
+
+	@Override
+	public @NotNull Mirror getMirror()
+	{
+		return this.mirror;
+	}
+
+	@Override
+	public void setRotation(@NotNull StructureRotation rotation)
+	{
+		Preconditions.checkNotNull(rotation, "Rotation cannot be null");
+		this.rotation = rotation;
+	}
+
+	@Override
+	public @NotNull StructureRotation getRotation()
+	{
+		return this.rotation;
+	}
+
+	@Override
+	public void setUsageMode(@NotNull UsageMode mode)
+	{
+		Preconditions.checkNotNull(mode, "Usage Mode cannot be null");
+		this.usageMode = mode;
+	}
+
+	@Override
+	public @NotNull UsageMode getUsageMode()
+	{
+		return this.usageMode;
+	}
+
+	@Override
+	public void setIgnoreEntities(boolean ignoreEntities)
+	{
+		this.ignoreEntities = ignoreEntities;
+	}
+
+	@Override
+	public boolean isIgnoreEntities()
+	{
+		return this.ignoreEntities;
+	}
+
+	@Override
+	public void setShowAir(boolean showAir)
+	{
+		this.showAir = showAir;
+	}
+
+	@Override
+	public boolean isShowAir()
+	{
+		return this.showAir;
+	}
+
+	@Override
+	public void setBoundingBoxVisible(boolean showBoundingBox)
+	{
+		this.showBoundingBox = showBoundingBox;
+	}
+
+	@Override
+	public boolean isBoundingBoxVisible()
+	{
+		return this.showBoundingBox;
+	}
+
+	@Override
+	public void setIntegrity(float integrity)
+	{
+		Preconditions.checkArgument(StructureMock.isBetween(integrity, 0.0f, 1.0f), "Integrity must be between 0 and 1");
+		this.integrity = integrity;
+	}
+
+	@Override
+	public float getIntegrity()
+	{
+		return this.integrity;
+	}
+
+	@Override
+	public void setSeed(long seed)
+	{
+		this.seed = seed;
+	}
+
+	@Override
+	public long getSeed()
+	{
+		return this.seed;
+	}
+
+	@Override
+	public void setMetadata(@NotNull String metadata)
+	{
+		Preconditions.checkNotNull(metadata, "Metadata cannot be null");
+		if (this.getUsageMode() == UsageMode.DATA)
+		{
+			this.metadata = metadata;
+		}
+	}
```

Comment:
Should we maybe warn the User here when the UsageMode is something other than Data

Reply:
No. [Here's](https://minecraft.fandom.com/wiki/Structure_Block#Usage) the wiki page on usage modes.